src/components/protocol_handler/src/protocol_packet.cc

```diff
@@ -379,29 +379,32 @@ RESULT_CODE ProtocolPacket::deserializePacket(
 
   packet_data_.totalDataBytes = packet_header_.dataSize;
 
-  uint32_t dataPayloadSize = 0;
-  if ((offset < messageSize) &&
-      packet_header_.frameType != FRAME_TYPE_FIRST) {
-    dataPayloadSize = messageSize - offset;
+  if (messageSize <= offset) {
+    return RESULT_OK;
   }
 
-  if (packet_header_.frameType == FRAME_TYPE_FIRST) {
-    payload_size_ = 0;
+  if (packet_header_.frameType != FRAME_TYPE_FIRST) {
+
```

Comment:
fixed in [4c1d58](https://github.com/LuxoftSDL/sdl_core/pull/15/commits/4c1d58eac9c6b78e9674aece3b615e9c96629ac6)